inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingService.java

```diff
@@ -397,22 +426,25 @@ private synchronized Sentence getMentionSentence(JCas aJcas, int aBegin)
         }
 
         // Set frequency
-        if (entityFrequencyMap != null) {
-            for (CandidateEntity l : aCandidatesFullText) {
-                String key = l.getIRI();
-                // For UKP Wikidata
-                if (aKB.getType() == RepositoryType.REMOTE
-                    && aKB.getFullTextSearchIri().equals(IriConstants.FTS_VIRTUOSO)) {
-                    RepositoryImplConfig cfg = kbService.getKnowledgeBaseConfig(aKB);
-                    if (((SPARQLRepositoryConfig) cfg).getQueryEndpointUrl()
-                        .equals(IriConstants.UKP_WIKIDATA_SPARQL_ENDPOINT)) {
-                        key = key.replace(IriConstants.PREFIX_WIKIDATA_ENTITY, "");
-                        if (entityFrequencyMap.get(key) != null) {
-                            l.setFrequency(entityFrequencyMap.get(key));
-                        }
+
+        if (aKB.getType() == RepositoryType.REMOTE && entityFrequencyMap != null) {
+            RepositoryImplConfig cfg = kbService.getKnowledgeBaseConfig(aKB);
+            if (((SPARQLRepositoryConfig) cfg).getQueryEndpointUrl()
+                .equals(IriConstants.UKP_WIKIDATA_SPARQL_ENDPOINT)) {    // For UKP Wikidata
+                for (CandidateEntity l : aCandidatesFullText) {
+                    String key = l.getIRI();
+                    key = key.replace(IriConstants.PREFIX_WIKIDATA_ENTITY, "");
+                    if (entityFrequencyMap.get(key) != null) {
+                        l.setFrequency(entityFrequencyMap.get(key));
                     }
                 }
             }
+        } else if (aKB.getType() == RepositoryType.LOCAL) {
+            for (CandidateEntity l : aCandidatesFullText) {
+                List<SearchResult> results 
+                    = getSearchResults(aUser, aKB.getProject(), l.getAltLabel());
+                l.setIdRank(results.size());
```

Comment:
I would rename `IdRank` to Popularity or so, as we now have more priors than Wiki id